launcher/ui/pages/instance/ModFolderPage.cpp

```diff
@@ -84,50 +84,69 @@ ModFolderPage::ModFolderPage(BaseInstance* inst, std::shared_ptr<ModFolderModel>
 
         connect(ui->actionDownloadItem, &QAction::triggered, this, &ModFolderPage::installMods);
 
+        // update menu
+        auto updateMenu = ui->actionUpdateItem->menu();
+        if (updateMenu) {
+            updateMenu->clear();
+        } else {
+            updateMenu = new QMenu(this);
+        }
+
+        {
+            auto update = updateMenu->addAction(tr("Check for Updates"));
+            update->setToolTip(tr("Try to check or update all selected mods (all mods if none are selected)"));
+            connect(update, &QAction::triggered, this, &ModFolderPage::updateMods);
+        }
+        if (!APPLICATION->settings()->get("ModDependenciesDisabled").toBool()) {  // dependencies
+
+            auto updateWithDeps = updateMenu->addAction(tr("Verify Dependencies"));
+            updateWithDeps->setToolTip(
+                tr("Try to update and check for missing dependencies all selected mods (all mods if none are selected)"));
+            connect(updateWithDeps, &QAction::triggered, this, [this] { updateMods(true); });
+        }
```

Comment:
Maybe make it sync with the setting e.g. (not tested)

```suggestion
        auto update = updateMenu->addAction(tr("Check for Updates"));
        update->setToolTip(tr("Try to check or update all selected mods (all mods if none are selected)"));
        connect(update, &QAction::triggered, this, &ModFolderPage::updateMods);

        auto updateWithDeps = updateMenu->addAction(tr("Verify Dependencies"));
        updateWithDeps->setToolTip(
            tr("Try to update and check for missing dependencies all selected mods (all mods if none are selected)"));
        connect(updateWithDeps, &QAction::triggered, this, [this] { updateMods(true); });

        auto depsDisabled = APPLICATION->settings()->getSetting("ModDependenciesDisabled");
        updateWithDeps->setVisible(!depsDisabled->get().toBool());
        connect(depsDisabled.get(), &Setting::SettingChanged, this,
                [](const Setting& setting, QVariant value) { updateWithDeps->setVisible(!value.toBool()); });
```